### PR TITLE
neovim, libuv: add debug output

### PR DESCRIPTION
**Pull request by symphorien**
###### Description of changes

both weigh about 7MB.
would have been nice to have for https://github.com/NixOS/nixpkgs/issues/218939

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
But we should wait for ofborg to be clean! See https://github.com/NixOS/nixpkgs/pull/219405

**Comment by wegank** [comment]
@ofborg eval

**Comment by figsoda** [comment]
this seems to break static build

**Comment by symphorien** [comment]
> this seems to break static build, see ofborg logs

This seems to be a more general problem with static libraries and separateDebugInfo. I opened an issue https://github.com/NixOS/nixpkgs/issues/219466
Meanwhile, I'll put this PR on hold.

**Comment by vcunat** [comment]
Well, if static matters in this case, perhaps simply do this for now?
```nix
separateDebugInfo = !stdenv.hostPlatform.isStatic;
```

**Comment by symphorien** [comment]
ok, done

**Review by figsoda — APPROVED** [review]

File: pkgs/development/libraries/libuv/default.nix [hunk 1]
```diff
@@ -74,6 +74,9 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  # https://github.com/NixOS/nixpkgs/issues/219466
```
Comment (teto) [inline pkgs/development/libraries/libuv/default.nix:77]: maybe prefix with some TLDR: .e.g, : `breaks static builds see ...` ?

**Comment by teto** [comment]
could @CnTeng test this to see if he indeed gets a better coredump ?

**Comment by symphorien** [comment]
> could @CnTeng test this to see if he indeed gets a better coredump ?

Note it's a mass rebuild and neovim depends on rustc, so it's a long build.
To use this debug info you need additional set up, for example https://github.com/symphorien/nixseparatedebuginfod

**Comment by CnTeng** [comment]
https://github.com/neovim/neovim/issues/21467
Maybe, this problem has been improved. After using gitgutter instead of gitsigns and switching to kernel 6.1, the crash rarely happened.
And my laptop performance is not good, so unfortunately I cannot review this. I am so sorry.

**Comment by symphorien** [comment]
I checked that:
- with just the first commit I get source file info when running `start` in gdb on neovim-unwrapped
- I get symbols for libuv with `info shared` in gdb on cmake

File: pkgs/development/libraries/libuv/default.nix (hunk 1 above)
Comment (symphorien) [reply pkgs/development/libraries/libuv/default.nix:78]: added

**Review by teto — APPROVED** [review]

**Comment by nullishamy** [comment]
> [neovim/neovim#21467](https://github.com/neovim/neovim/issues/21467) Maybe, this problem has been improved. After using gitgutter instead of gitsigns and switching to kernel 6.1, the crash rarely happened. And my laptop performance is not good, so unfortunately I cannot review this. I am so sorry. @nullishamy has the same problem, maybe she can give some help.

I'd be down to give it a go. How exactly do you get this version to run locally? I'm new to doing nixpkgs development in this fashion haha!

**Comment by symphorien** [comment]
with `git-extras` installed:
```
git clone https://github.com/NixOS/nixpkgs
cd nixpkgs
git pr https://github.com/NixOS/nixpkgs/pull/219400
nix-build -A neovim
```
the compiled neovim is `./result/bin/nvim`

**Comment by nullishamy** [comment]
Thanks for that. Got it running, and still getting the crashes with the same function:
```
#0  0x00007f79dfeaa9ee in push_fs_result () from /nix/store/gw7w2qb2afjclxpc4hqj0hjzqv807fic-libluv-1.43.0-0/lib/libluv.so.1
```

**Comment by nullishamy** [comment]
I did not. I can rerun with that and report back.

**Comment by teto** [comment]
according to this comment https://github.com/luvit/luv/issues/644#issuecomment-1500989929 , the PR seems to work. Let's merge.

**Pull request merged**